### PR TITLE
Update aad_group_name resolution

### PR DESCRIPTION
**Pull request by endakelly**
### Change description ###
local value is not being resolved correctly. Currently being resolved to sharedservices related groups in aks-cft-deploy
[Link](https://dev.azure.com/hmcts/PlatformOperations/_build/results?buildId=388198&view=logs&j=fbe5e610-086d-53cd-fb9a-f5a3d2cb4fdf&t=aa697d24-2372-5379-79e2-23425abbeabc) to pipeline run with this change implemented.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: 03-interpolated-defaults.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
 locals {
 
   sub_env        = var.environment == "ptlsbox" ? "ptl-sbox" : var.environment
-  aad_group_name = contains(["ptlsbox", "ptl"], var.environment) ? "DTS Contributors (sub:dts-sharedservices${local.sub_env})" : "DTS Contributors (sub:dts-sharedservices-${local.sub_env})"
-
   
-  project = length(regexall(".*sds.*", lower(var.developers_group))) > 0 ? "sds" : "cft"
+  aad_group_name = var.environment == "ptl" && local.project == "cft" ? "DTS Contributors (sub:dts-cftptl-intsvc)" : (var.environment == "ptlsbox" && local.project == "cft" ? "DTS Contributors (sub:dts-cftsbox-intsvc)" : var.environment == "ptl" && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservicesptl)" : (var.environment == "ptlsbox" && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservicesptl-sbox)" : (!contains(["ptl","ptlsbox"], var.environment) && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservices-${local.sub_env})" : "DTS Contributors (sub:dcd-cftapps-${local.sub_env})")))
```
Comment (timja) [inline 03-interpolated-defaults.tf:5]: how was this working before?

I can't see any CFT specific logic.

Anyway isn't this just the active subscriptions displayname lowercased?
You could get the exact algorithm from the azure-enterprise repository

**Review by timja — CHANGES_REQUESTED** [review]

File: 03-interpolated-defaults.tf (hunk 1 above)
Comment (endakelly) [reply 03-interpolated-defaults.tf:5]: The project local value was being derived from the variable developers group which, in aks-cft-deploy, is set to  `DTS Operations env: $env` which has no mention of `cft` or `sds` so this local always resolves to `sds`. This local is used to determine the keyvault name [here](https://github.com/hmcts/aks-module-genesis/blob/7af7c076fbe1e3db172b1b819360ca88ad6d090a/20-key-vault.tf#L7).

The aad_group_name local currently has sharedservices groups hardcoded so, at the minute, cft keyvaults have the sharedservices contributors groups granted access but not the equivalent groups for cft. Not sure why it was left this way.

Comment (timja) [reply 03-interpolated-defaults.tf:5]: I would start by changing the project variable to be `businessArea` and then it should be passed down from the consuming repositories.

I think you should be able to determine the group name sensibly then

Comment (endakelly) [reply 03-interpolated-defaults.tf]: Tried to simplify it somewhat using businessArea tag. Getting tag value from https://github.com/hmcts/aks-cft-deploy/pull/457/files#diff-0aa6a4ae14301983310f5a332fd11d5c51ed8c5848613358af32d90599a80331R6

File: 03-interpolated-defaults.tf [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+    "sds-ptl" = {
+      "DTS Contributors (sub:dts-sharedservicesptl)" = {}
+    }
+    "sds-ptlsbox" = {
+      "DTS Contributors (sub:dts-sharedservicesptl-sbox)" = {}
+    }
+    "sds-env" = {
+      "DTS Contributors (sub:dts-sharedservices-${local.sub_env})" = {}
+    }
+  }
```
Comment (timja) [inline 03-interpolated-defaults.tf:26]: can't we do:

```terraform
locals {
  contributors_group = {
    "cft-ptl" = "DTS Contributors (sub:dts-cftptl-intsvc)"
    // ...
  }
}
```

File: 03-interpolated-defaults.tf [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+
+  business_area = var.business_area == "cft" ? var.business_area : "sds"
```
Comment (timja) [inline 03-interpolated-defaults.tf:28]: we generally pass through sds instead of cross cutting I think so I don't think you would need this

Comment (endakelly) [reply 03-interpolated-defaults.tf:16]: Seems we're passing cross-cutting in [aks-sds-deploy](https://dev.azure.com/hmcts/PlatformOperations/_build/results?buildId=388970&view=logs&j=fbe5e610-086d-53cd-fb9a-f5a3d2cb4fdf&t=aa697d24-2372-5379-79e2-23425abbeabc&l=405)

Comment (timja) [reply 03-interpolated-defaults.tf:16]: ok in that case I would suggest overriding the specific one we know about and not special casing cft as that doesn't seem right
```suggestion
  business_area = var.business_area == "cross-cutting" ? "sds : var.business_area
```

**Review by timja — APPROVED** [review]

File: 01-inputs-required.tf [hunk 4]
```diff
@@ -1,2 +1,3 @@
 variable "environment" {}
 variable "developers_group" {}
+variable "business_area" {}
```
Comment (timja) [inline 01-inputs-required.tf:3]: not for now but this module should really be documented better 😉

**Pull request merged**